🪟 🚥 Add tests for verify sync mode options

SofiiaZaitseva opened the pull request:
Add auto tests for checking Sync modes for Stream:

Full Refresh | Overwrite
Full Refresh | Append
Incremental | Append
Incremental | Deduped History
Test cases (c13-c18)
Link to doc with TCs: https://docs.google.com/spreadsheets/d/1OMKYYoTAFeCAk_ZXUlFmiKpfnt7x5edRGxRQB4XMyQM/edit?userstoinvite=leti@airbyte.io&actionButton=1#gid=979953315

SofiiaZaitseva committed:
add tests for verify sync mode options

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=16549) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=16549) before we can accept your contribution.<br/>**10** out of **11** committers have signed the CLA.<br/><br/>:white_check_mark: suhomud<br/>:white_check_mark: dizel852<br/>:white_check_mark: sophia-wiley<br/>:white_check_mark: ambirdsall<br/>:white_check_mark: SofiiaZaitseva<br/>:white_check_mark: git-phu<br/>:white_check_mark: xpuska513<br/>:white_check_mark: DoNotPanicUA<br/>:white_check_mark: grubberr<br/>:white_check_mark: girarda<br/>:x: github-actions[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=16549) it.</sub>

krishnaglick commented on airbyte-webapp-e2e-tests/cypress/commands/connection.ts:
```diff
@@ -13,13 +13,23 @@ export const createTestConnection = (sourceName: string, destinationName: string
       break;
     case sourceName.includes('Postgres'):
-      createPostgresSource(sourceName);
+      createPostgresSource(sourceName, "ec2-3-229-11-55.compute-1.amazonaws.com", "{selectAll}{del}5432", "d5pcodgtmh9pob", "yuxothkqynerju", "1c1771f795985693bc4f780130d7e8af895a3d48155007b93d21218e4bc07cd9", "demo");
```
Is this a PAT? CC @timroes

timroes replied:
We have integration test data that we somehow inject into integration tests for connectors.

There are two options here, we can either use a hosted Postgres here as well and inject that data in a similar matter, which has the disadvantage that this won't work for OSS users anymore, since only we can run those tests in our infrastructure. Alternatively, we could just export a SQL dump from the data we want and make sure we inject that before running our tests into our locally running Postgres so we always have the same data, and not use hosted test instances unless really needed (because we might want to test an API service).

@ambirdsall I'd refer to you here once you're back for more clarification and setting directions.

ambirdsall replied:
Unless there's some context I'm missing, I have a pretty solid preference for the second solution (using a SQL dump to insert known test data in a local postgres database) to keep this working as well as possible for all external open-source contributors. It would be fairly straightforward to define a `"postcreatedb"` npm script to load the SQL dump; this would be automatically run after `npm run createdb`. I haven't had a chance to test this yet, but I would expect changing `airbyte-webapp-e2e-tests/package.json` with something like the following would work:
```diff
-    "createdb": "docker run --rm -d -p 5433:5432 -e POSTGRES_PASSWORD=secret_password -e POSTGRES_DB=airbyte_ci --name airbyte_ci_pg postgres"
+    "createdb": "docker run --rm -d -p 5433:5432 -e POSTGRES_PASSWORD=secret_password -e POSTGRES_DB=airbyte_ci --name airbyte_ci_pg postgres",
+    "postcreatedb": "docker run --rm -d -p 5433:5432 -e POSTGRES_PASSWORD=secret_password -e POSTGRES_DB=airbyte_ci --name airbyte_ci_pg psql --single-transaction < test-data.sql"
},
```

Do we have any documentation or reference for the desired test data? If not, dumping the remote database that is currently configured in this PR seems like a reasonable starting point.

krishnaglick commented on airbyte-webapp-e2e-tests/cypress/commands/connector.ts:
```diff
@@ -1,18 +1,19 @@
   enterHost(host);
   enterPort(port);
   enterDatabase(database);
+  if (schema != "") {
```
```suggestion
  if (schema !== "") {
```
With Lake's linting PR it would likely flag this.

krishnaglick reviewed (commented):
Please merge in master as I think the linter may have some opinions.

SofiiaZaitseva committed:
Update airbyte-webapp-e2e-tests/cypress/commands/connector.ts

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

SofiiaZaitseva committed:
fix selector

fix selector

SofiiaZaitseva committed:
fix naming and locator

fix naming and locator

SofiiaZaitseva committed:
fix formatting

ambirdsall commented on airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts:
```diff
@@ -62,41 +66,125 @@ describe("Connection main actions", () => {
+    submitButtonClick();
+    confirmStreamConfigurationChangedPopup();
+
+    cy.wait(5000);
```
This `cy.wait(5000)` is [not necessary](https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting): if you see timeout errors on the next line's `cy.wait("@updateConnection").then(/* test assertion */);` call, you can add the extra timeout to it directly by adding an addition options argument, like `cy.wait("@updateConnection", { timeout: 5000 }).then(/* test assertion */);`. You can safely make the timeout in this options argument very long: because it's part of the wait _for a specific network call_, cypress will stop waiting as soon as it sees the call, no matter how much time is left.

ambirdsall commented on airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts:
```diff
@@ -62,41 +66,125 @@ describe("Connection main actions", () => {
+
+    searchStream("admins");
+
+    checkCursorField("email");
```
Given that the test is named "Connection sync mode Incremental Append", this code should check the sync mode field and not the cursor field. In fact, as written there is no difference between the assertions in this test and in the following one: in fact, if there were a backend bug which saved every connection with `Full refresh | Append` sync mode (exactly the kind of error this test should fail on!), both tests would still pass.

SofiiaZaitseva replied:
Actually we cannot same Full refresh | Append sync mode without cursor field so it looks like as a necessary details.

ambirdsall commented on airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts:
```diff
@@ -42,12 +45,28 @@ export const setupDestinationNamespaceSourceFormat = () => {
+};
+
+export const checkPrimaryKey = (expectedValue: string) => {
+  cy.get(primaryKeyText).should("have.text", expectedValue);
```
The same goes for this and other test code which uses `cy.get(someSelector).should("have.text", expectedValue)`: `cy.contains(expectedValue)` or `cy.get(entirePageSectionForSomeSpecificRecord).contains(expectedValue)` is better since it lets us improve or fix the code more easily without needing to edit the test code too.

SofiiaZaitseva committed:
comment's fixes

SofiiaZaitseva committed:
add tests for checking destination namespace (#18027)

add tests for checking destination namespace

dizel852 committed:
add cypress-postgres package

dizel852 committed:
fix plugins/index.ts warnings

dizel852 committed:
fix plugins/index.ts warnings

dizel852 committed:
setup test db actions

dizel852 committed:
add setup db (just for testing purpose)

dizel852 committed:
add create db command to gradle

dizel852 committed:
Revert "add create db command to gradle"

This reverts commit bc0f350fbd603a898400e2441e3da80eae097cc8.

dizel852 committed:
add test-id to components

dizel852 committed:
update db queries in tests

dizel852 committed:
fix format

dizel852 committed:
add tests with diff modal

dizel852 committed:
remove 'only' flag

dizel852 committed:
Merge branch 'master' into e2e-diff-modal-test-cases

# Conflicts:
#	airbyte-webapp/src/views/Connection/CatalogDiffModal/useConfirmCatalogDiff.tsx

dizel852 committed:
update failed snapshots

dizel852 committed:
update aria label and data-testid in components

dizel852 committed:
move selectors to separate pages

dizel852 committed:
update tests

dizel852 committed:
update test snapshots

dizel852 committed:
update test since data-testid was added

dizel852 committed:
Merge branch 'master' into e2e-diff-modal-test-cases

dizel852 committed:
Merge branch 'master' into e2e-diff-modal-test-cases

suhomud committed:
17644 Update Destination data type test to use the new data types (#19281)

This PR only changes tests and does not affect production, so it is safe to merge. 

* 17644 Destination acceptance test: protocol support

* 17644 data type array test catalog update

* 17644 correct array string type

* add v0 and v1 versions

* added binary_data type

* Added infinity, nan support. Added test time dataset. Updated ref link

* format fix

ambirdsall committed:
:window: :tada: Display service token validation errors in the UI (#19578)

Note: the backend returns a 500 status (should be 400, or at least 4xx), so the frontend has to remove the error message's unsightly "Internal Server Error: " prefix in post.

In the future, we want to move this to a 400 error.

sophia-wiley committed:
dbt Cloud integration doc (#19619)

dbt Cloud integration doc

Made some changes since we last reviewed.

- Deleted the sections about how to find your account ID and job ID because the integration now fetches those for you after you add your service token.
- Deleted the section about viewing logs.
Edited to reflect the changes to the + Add transformation dropdown.

girarda committed:
Add connector builder server to airbyte proxy, kube overlays, and helm charts (#19554)

github-actions[bot] committed:
Bump helm chart version reference to 0.41.3 (#19685)

Co-authored-by: sherifnada <sherifnada@users.noreply.github.com>

octavia-squidington-iii committed:
Bump Airbyte version from 0.40.21 to 0.40.22 (#19687)

Co-authored-by: lmossman <lmossman@users.noreply.github.com>
Co-authored-by: lmossman <lake@airbyte.io>

xpuska513 committed:
fix: add extraEnv block (#19703)

* fix: add extraEnv block

* fix nindent

github-actions[bot] committed:
Bump helm chart version reference to 0.42.0 (#19706)

Co-authored-by: xpuska513 <xpuska513@users.noreply.github.com>

DoNotPanicUA committed:
fix order not to randomly fail backward compatibility check (#19377)

* fix order not to randomly fail backward compatibility check

* missing file update

Commit:
🎉 Source Asana: migrate to new SAT, added base HTTP errors handling (#19561)

dizel852 commented:
Synced with @SofiiaZaitseva, what was discussed::
- Use pattern for table creation from [PR](https://github.com/airbytehq/airbyte/pull/19447) 
- Add create destination DB in bash file in order to run multiple DBs in CI `(/Users/user/Desktop/airbyte/tools/bin/e2e_test.sh)`
- Update package.json to have the commands `createDBSource` and `createDBDestination` (Local run)
- Cover cases when `PRIMARY KEY` is defined, and not defined(multiple `PK` selections are available)

cc: @timroes

grubberr committed:
Source Google Ads: fix schema for "campaigns" stream (#19700)

Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

git-phu committed:
find_non_rate_limited_PAT (#19736)

* find_non_rate_limited_PAT

* checkout repo to find pat

Commit:
fix typo: notify instead of sync (#19737)

grubberr committed:
Source facebook marketing: check "breakdowns" combinations (#19645)

Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

Commit:
🪟 🐛 Fix custom connection creation endpoint (#19702)

Closes #19637

PR that introduced this bug was a precision revert of another issue: #19627

This PR removes the createSourceDefinition() and createDestinationDefinition() methods in favor of the new createCustomSourceDefinition() and createCustomDestinationDefinition() methods. createSourceDefinition() and createDestinationDefinition() are to be deprecated.

This PR does not address updating custom connectors. There is a backend issue for fixing this behavior here: #19669

Commit:
fix: yaml syntax (#19775)

xpuska513 committed:
fix: add BuildPulse report for helm ac tests (#19785)

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
add implementation of using local db for checking sync mode

SofiiaZaitseva committed:
fix creation db in docker

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
add test for checking sync mode Incremental Deduped History without existing PK

dizel852 reviewed (changes requested):
Code LGTM, but need to make a few changes and add comments to description.
What I've noticed:
- This PR is based on another one(please, provide link), since reviewers will be confused why they see the same changes in both PRs
- List all the test cases we cover, with IDs from the doc we have atm(would be great to provide link on the doc as well )
- The PR(on changes this one is based) should be reviewed and merged at first, thereafter we won't see the changes form the PR we used in this one. This will make the review process more slick.
- Need to update PR dependency

SofiiaZaitseva committed:
Update airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts

Co-authored-by: Vladimir <volodymyr.s.petrov@globallogic.com>

SofiiaZaitseva committed:
Update airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts

Co-authored-by: Vladimir <volodymyr.s.petrov@globallogic.com>

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
Update airbyte-webapp-e2e-tests/package.json

Co-authored-by: Vladimir <volodymyr.s.petrov@globallogic.com>

SofiiaZaitseva committed:
fix db naming

SofiiaZaitseva commented on airbyte-webapp-e2e-tests/cypress/commands/connection.ts:
```diff
@@ -12,14 +12,25 @@ export const createTestConnection = (sourceName: string, destinationName: string
-  createLocalJsonDestination(destinationName, "/local");
+  switch (true) {
+    case destinationName.includes("Postgres"):
+      createPostgresDestination(destinationName, "localhost", "5434", "airbyte_ci_destination");
```
done

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
fix e2e run

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests

SofiiaZaitseva committed:
fix db name

SofiiaZaitseva committed:
Merge branch 'Sofiia_AddStatusConnectionPageTests' of https://github.com/airbytehq/airbyte into Sofiia_AddStatusConnectionPageTests

dizel852 reviewed (approved):
Reviewed after fixes - all 👍 .
Passing to @edmundito for another review

SofiiaZaitseva committed:
Merge branch 'master' into Sofiia_AddStatusConnectionPageTests
